src/lib.rs

```diff
@@ -106,6 +118,15 @@ pub enum AstarteError {
 
     #[error("conversion error")]
     Conversion,
+
+    #[error("impossible error")]
+    Impossible,
+}
+
+impl From<Infallible> for AstarteError {
+    fn from(_: Infallible) -> Self {
+        AstarteError::Impossible
+    }
```

Comment:
Converting to an `AstarteError` enum does not need to use the `from` trait, the enum uses the `derive thiserror` macro that allows you to write:
``` Rust
#[error("infallible error")]
Infallible(#[from] std::convert::Infallible)
```    
Add this lines to the AstarteError enum and remove the `from` trait.